pkg/conversion/mapi2capi/aws_test.go

```diff
@@ -16,43 +16,140 @@ limitations under the License.
+		{NoDevice: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
+	awsMAPIMachineWithVirtualName = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
+	awsMAPIMachineWithVirtualNameAndRootVolNotDeletedOnTerm = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10)), DeleteOnTermination: ptr.To(false)}},
+	}))
+	awsMAPIMachineWithVirtualNameAndMissingEBSConfig = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test")},
+	}))
+	awsMAPIMachineWithMissingEBSConfig = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{DeviceName: ptr.To("/dev/sdb")},
+	}))
+	awsMAPIMachineWithRootVolumeNotDeleted = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10)), DeleteOnTermination: ptr.To(false)}},
+	}))
+	awsMAPIMachineWithNoDeviceAndVirtualName = awsMAPIMachineBase.WithProviderSpecBuilder(awsBaseProviderSpec.WithBlockDevices([]mapiv1.BlockDeviceMappingSpec{
+		{VirtualName: ptr.To("test"), NoDevice: ptr.To("test"), EBS: &mapiv1.EBSBlockDeviceSpec{VolumeSize: ptr.To(int64(10))}},
+	}))
+
+	// MachineSets configurations.
+	awsMAPIMachineSetBase = machinebuilder.MachineSet().WithProviderSpecBuilder(awsBaseProviderSpec)
+
+	// OCP Infrastructure configurations.
+	infra = &configv1.Infrastructure{
+		Spec:   configv1.InfrastructureSpec{},
+		Status: configv1.InfrastructureStatus{InfrastructureName: "sample-cluster-name"},
+	}
+	infraWithRegion = configbuilder.Infrastructure().AsAWS("sample-cluster-name", awsDefaultRegion)
 )
 
-var _ = Describe("mapi2capi AWS", func() {
-	awsMAPIMachine := machinebuilder.Machine().WithProviderSpecBuilder(machinebuilder.AWSProviderSpec()).WithProviderID("aws:///us-west-2a/i-05442bc41c3df969d").Build()
-	awsMAPIMachineSet := machinebuilder.MachineSet().WithProviderSpecBuilder(machinebuilder.AWSProviderSpec()).Build()
-	awsMAPIMachineSet.Spec.Template.Spec.ProviderID = ptr.To("aws:///us-west-2a/i-05442bc41c3df969d") // TODO: do this in machinebuilder.
-	infra := configv1.Infrastructure{
-		Spec: configv1.InfrastructureSpec{},
-		Status: configv1.InfrastructureStatus{
-			InfrastructureName: "sample-cluster-name",
-		},
+var _ = DescribeTable("mapi2capi AWS convert MAPI Machine",
+	func(m *mapiv1.Machine, i *configv1.Infrastructure, errorsMatcher types.GomegaMatcher, warningsMatcher types.GomegaMatcher) {
```

Comment:
Please use an input struct rather than named positional arguments like this, it makes the entries _sooo_ much easier to read.

I'd expect the entries to look like
```
Entry("name", awsConversionInput{
  machine: awsMAPIMachineBase.Build(),
  Infra: infra,
  expectedError: "...", <- Typical pattern is to handle this within the table
  expectedWarnings: []string{},
}),
```

In the past we have even done the construction of the providerSpec differences within the Entry, so that you can see exactly the changes here, rather than having to refer up to a particular variable above

Reply:
I'm thinking we want these more like [this example](https://github.com/openshift/cluster-control-plane-machine-set-operator/blob/8d85b2cb2c1dc8709fa49470b1df91640c642a59/pkg/controllers/controlplanemachineset/updates_test.go#L111) in CPMS where in Each `Entry` you have everything you need to be able to understand the complete test input, WDYT?

Reply:
Yes good idea, done.